stdlib/public/core/StringBridge.swift

```diff
@@ -427,34 +459,33 @@ private func getConstantTaggedCocoaContents(_ cocoaString: _CocoaString) ->
 
   let taggedValue = unsafeBitCast(cocoaString, to: UInt.self)
   
-  //11000000..payload..111
-  let tagMask:UInt =
-    0b1111_1111_1000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0111
-  let expectedValue:UInt =
-    0b1100_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0111
 
-  guard taggedValue & tagMask == expectedValue else {
-    return nil
-  }
 
-  guard _swift_stdlib_dyld_is_objc_constant_string(
-    cocoaString as! UnsafeRawPointer
-  ) == 1 else {
+  guard taggedValue & constantTagMask == expectedConstantTagValue else {
     return nil
   }
 
-  let payloadMask = ~tagMask
+  let payloadMask = ~constantTagMask
   let payload = taggedValue & payloadMask
   let ivarPointer = UnsafePointer<_swift_shims_builtin_CFString>(
     bitPattern: payload
   )!
+
+  guard _swift_stdlib_dyld_is_objc_constant_string(
+    unsafeBitCast(ivarPointer, to: UnsafeRawPointer.self)
```

Comment:
I wonder if going through `ObjectIdentifier` is more sanctioned. @lorentey , thoughts?